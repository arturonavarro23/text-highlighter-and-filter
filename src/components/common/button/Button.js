import React from 'react';
import classnames from 'classnames';

import './Button.scss';

const Button = ({ variant, onClick, selected }) => {
  const btnClass = classnames({
    btn: true,
    [`btn--${variant}`]: true,
    selected,
  });

  return (
    <button
      onClick={onClick}
      className={btnClass}
    />
  );
};

export default Button;
