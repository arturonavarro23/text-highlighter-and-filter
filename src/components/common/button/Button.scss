@import '../../../scss/abstracts/variables';

.btn {
  cursor: pointer;
  margin-right: 15px;
  width: 40px;
  height: 40px;
  border-radius: 50%;

  &.btn--red {
    background-color: $red-color;
  }

  &.btn--green {
    background-color: $green-color;
  }

  &.btn--yellow {
    background-color: $yellow-color;
  }

  &.selected {
    border: solid 3px $blue-color;
  }
}
