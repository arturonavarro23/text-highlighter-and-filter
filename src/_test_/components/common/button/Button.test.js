import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '../../../../components/common/button';

describe('<ColorSelector> test', () => {
  it('Snapshot should renders correctly', () => {
    const mockFn = jest.fn();
    const button = (
      <Button
        variant="red"
        onClick={mockFn}
      />
    );

    const colorSelectorTree = renderer
      .create(button)
      .toJSON();

    expect(colorSelectorTree).toMatchSnapshot();
  });
  it('Should be selected', () => {
    const mockFn = jest.fn();

    const buttonWrapper = mount(
      <Button
        variant="red"
        onClick={mockFn}
        selected
      />
    );

    expect(buttonWrapper.find('button').hasClass('btn--red')).toBe(true);
  });
  it('onClick should be called', () => {
    const mockFn = jest.fn().mockName('onClick');

    const buttonWrapper = mount(
      <Button
        variant="red"
        onClick={mockFn}
        selected
      />
    );

    const button = buttonWrapper.find('button');
    button.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});
