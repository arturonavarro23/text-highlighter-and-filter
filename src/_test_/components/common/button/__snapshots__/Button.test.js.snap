// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorSelector> test Snapshot should render correctly 1`] = `
<button
  className="btn btn--red"
  onClick={[MockFunction]}
/>
`;
